pkg/controllers/management/kontainerdrivermetadata/metadata_handler.go

```diff
@@ -0,0 +1,164 @@
+		Addons:                    mgmt.RKEAddons(""),
+		WindowsSystemImagesLister: mgmt.RKEK8sWindowsSystemImages("").Controller().Lister(),
+		WindowsSystemImages:       mgmt.RKEK8sWindowsSystemImages(""),
+		ctx:                       ctx,
+	}
+
+	// load values on startup and start ticker
+	m.refresh(settings.RkeMetadataURL.Get())
+	m.initTicker(ctx)
+
+	mgmt.Settings("").AddHandler(ctx, "rke-metadata-handler", m.sync)
+}
+
+func (m *MetadataController) sync(key string, test *v3.Setting) (runtime.Object, error) {
+	if test == nil || (test.Name != "rke-metadata-url" && test.Name != "rke-metadata-refresh-interval") {
+		return nil, nil
+	}
+	url, interval, err := getSettingValues()
+	if err != nil {
+		return nil, fmt.Errorf("driverMetadata error getting interval in int %s %v", settings.RkeMetadataRefreshInterval.Get(), err)
+	}
+	if tickerData != nil {
+		if tickerData.url != url || tickerData.interval != interval {
+			tickerData.cancelFunc()
+
+			logrus.Infof("driverMetadata: starting new %s %v", url, interval)
+			cctx, cancel := context.WithCancel(m.ctx)
+			tickerData.interval = interval
+			tickerData.url = url
+			tickerData.cancelFunc = cancel
+
+			go m.startTicker(cctx, tickerData)
+		}
+	}
+	return test, nil
+}
+
+func (m *MetadataController) initTicker(ctx context.Context) {
+	url, interval, err := getSettingValues()
+	if err != nil {
+		panic(fmt.Errorf("driverMetadata error getting interval in int %s %v", settings.RkeMetadataRefreshInterval.Get(), err))
+	}
+	cctx, cancel := context.WithCancel(ctx)
+
+	tickerData = &TickerData{cancelFunc: cancel, url: url, interval: interval}
+
+	go m.startTicker(cctx, tickerData)
+}
+
+func (m *MetadataController) startTicker(ctx context.Context, tickerData *TickerData) {
+	checkInterval, url := tickerData.interval, tickerData.url
+	for range ticker.Context(ctx, checkInterval) {
+		logrus.Infof("driverMetadata: checking rke-metadata-url every %v", checkInterval)
+		m.refresh(url)
+	}
+}
+
+func (m *MetadataController) refresh(url string) {
+	data, err := loadData(url)
+	if err != nil {
+		logrus.Errorf("error loading rke data, using stored defaults %v", err)
+		if err := m.createOrUpdateMetadataDefaults(); err != nil {
+			logrus.Errorf("driverMetadata %v", err)
+		}
+		return
+	}
+
+	if err := m.createOrUpdateMetadata(data); err != nil {
+		logrus.Errorf("driverMetadata %v", err)
+	}
+}
+
+func getSettingValues() (string, time.Duration, error) {
+	n, err := strconv.Atoi(settings.RkeMetadataRefreshInterval.Get())
+	if err != nil {
+		return "", 0, err
+	}
+	checkInterval := time.Duration(n) * time.Minute
+	return settings.RkeMetadataURL.Get(), checkInterval, nil
+}
+
+func loadData(url string) (Data, error) {
+	var data Data
+	resp, err := http.Get(url)
+	if err != nil {
+		return data, fmt.Errorf("driverMetadata err %v", err)
+	}
+	if resp.StatusCode != 200 {
+		return data, fmt.Errorf("driverMetadata statusCode %v", resp.StatusCode)
+	}
+	defer resp.Body.Close()
```

Comment:
If resp.StatusCode!= 200, we would return without closing resp.Body; we should move this statement up